### PR TITLE
removing es6 support in ecs 8.0

### PR DESCRIPTION
**Pull request by kgeller**
Removing support of ES 6 in ECS 8.0 as it is 2 major versions apart as shown by the [support matrix](https://www.elastic.co/support/matrix).

### CODE REVIEW

**Comment by kgeller** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by kgeller** [comment]
@elasticmachine run elasticsearch-ci/docs

File: CHANGELOG.next.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ Thanks, you're awesome :-) -->
 
 #### Breaking changes
 
+* Remove template generation for ES6. #1680
```
Comment (ebeahan) [inline CHANGELOG.next.md:34]: If we backport to 8.0, let's move this under the 8.0.0 changelog.

File: scripts/generators/es_template.py [hunk 2]
```diff
@@ -286,30 +269,3 @@ def default_mapping_settings():
...
-    Visits each leaf in mappings object and fallback to an
-    Elasticsearch 6.x supported type.
-
-    Since a field like `wildcard` won't have the same defaults as
-    a `keyword` field, we must add any missing defaults.
-    """
-
-    for (name, details) in mappings.items():
-        if 'type' in details:
-            fallback_type = TYPE_FALLBACKS.get(details['type'])
```
Comment (ebeahan) [inline scripts/generators/es_template.py:302]: Is anything still using `TYPE_FALLBACKS`?

If not, that global can be cleared out.

File: scripts/generators/es_template.py [hunk 3]
```diff
@@ -286,30 +269,3 @@ def default_mapping_settings():
...
-                mappings[name]['type'] = fallback_type
-                field_or_multi_field_datatype_defaults(mappings[name])
-        # support multi-fields
-        if 'fields' in details:
-            # potentially multiple multi-fields
-            for field_name, field_value in details['fields'].items():
-                fallback_type = TYPE_FALLBACKS.get(field_value['type'])
-                if fallback_type:
-                    mappings[name]['fields'][field_name]['type'] = fallback_type
-                    field_or_multi_field_datatype_defaults(mappings[name]['fields'][field_name])
```
Comment (ebeahan) [inline scripts/generators/es_template.py:313]: This import from `schema.cleaner` might be removable `field_or_multi_field_datatype_defaults` now.

File: scripts/generators/es_template.py [hunk 4]
```diff
@@ -218,23 +217,7 @@ def template_settings(es_version, ecs_version, mappings_section, template_settin
     else:
         template = default_template_settings(ecs_version)
 
-    if es_version == 6:
-        mappings_section = copy.deepcopy(mappings_section)
```
Comment (ebeahan) [inline scripts/generators/es_template.py:222]: I think `copy` is now another unused import too.

**Review by ebeahan — CHANGES_REQUESTED** [review]

**Review by ebeahan — APPROVED** [review]
Thanks for making the changes - LGTM! 🚀

**Pull request merged**